File: src/main/java/no/fint/provider/springer/model/PersonRepository.java
package no.fint.provider.springer.model;

import lombok.extern.slf4j.Slf4j;
import no.fint.event.model.Event;
import no.fint.event.model.ResponseStatus;
import no.fint.event.model.Status;
import no.novari.fint.model.felles.FellesActions;
import no.novari.fint.model.resource.FintLinks;
import no.novari.fint.model.resource.felles.KontaktpersonResource;
import no.novari.fint.model.resource.felles.PersonResource;
import no.fint.provider.springer.storage.SpringerRepository;
import org.springframework.stereotype.Repository;

import java.util.Set;
import java.util.stream.Collectors;
import java.util.stream.Stream;

@Slf4j
@Repository
public class PersonRepository extends SpringerRepository {

    @Override
    public void accept(Event<FintLinks> response) {
        switch (FellesActions.valueOf(response.getAction())) {
            case GET_ALL_PERSON:
                query(PersonResource.class, response);
                break;
            case GET_ALL_KONTAKTPERSON:
                query(KontaktpersonResource.class, response);
                break;
            case UPDATE_PERSON:
            default:
                response.setStatus(Status.ADAPTER_REJECTED);
                response.setResponseStatus(ResponseStatus.REJECTED);
                response.setStatusCode("INVALID_ACTION");
                response.setMessage("Invalid action");
        }
    }

    @Override
    public Set<String> actions() {
        return Stream
                .of(FellesActions.values())
                .map(Enum::name)
                .filter(s -> s.startsWith("GET_ALL_"))
                .collect(Collectors.toSet());
    }

}

File: src/main/kotlin/no/fint/provider/springer/model/PersonRepository.kt
package no.fint.provider.springer.model

import no.fint.event.model.Event
import no.fint.event.model.ResponseStatus
import no.fint.event.model.Status
import no.fint.provider.springer.storage.SpringerRepository
import no.fint.provider.springer.storage.Wrapper
import no.novari.fint.model.felles.FellesActions
import no.novari.fint.model.resource.FintLinks
import no.novari.fint.model.resource.felles.KontaktpersonResource
import no.novari.fint.model.resource.felles.PersonResource
import org.springframework.data.mongodb.core.MongoTemplate
import org.springframework.stereotype.Repository
import java.util.stream.Collectors
import java.util.stream.Stream

@Repository
class PersonRepository(
    mongoTemplate: MongoTemplate,
    wrapper: Wrapper
) : SpringerRepository(mongoTemplate, wrapper) {

    override fun accept(response: Event<FintLinks>) {
        when (FellesActions.valueOf(response.action)) {
            FellesActions.GET_ALL_PERSON -> query(PersonResource::class.java, response)
            FellesActions.GET_ALL_KONTAKTPERSON -> query(KontaktpersonResource::class.java, response)
            else -> {
                response.status = Status.ADAPTER_REJECTED
                response.responseStatus = ResponseStatus.REJECTED
                response.statusCode = "INVALID_ACTION"
                response.message = "Invalid action"
            }
        }
    }

    override fun actions(): Set<String> =
        Stream.of(*FellesActions.values())
            .map { it.name }
            .filter { it.startsWith("GET_ALL_") }
            .collect(Collectors.toSet())
}
